### PR TITLE
feat: Smaller bundle option

### PR DESCRIPTION
**Pull request by benjackwhite**
## Changes

Response to someone asking about a reduced bundle size - attempting to see how hard it would be to strip the core down

## TODO
- [ ] Fix tests
- [ ] Move segment integration into extended
- [ ] Fix types for loaded event

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers
- [ ] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

**Review by pauldambra — COMMENTED** [review]
😍 

(very quick pass - shout when it's ready for some testing 🙌)

File: src/loaders/loader-array-full.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Same as loader-globals.ts except includes rrweb scripts.
+
+import './loader-recorder'
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply src/loaders/loader-array-full.ts:3]: _maybe_ probably yes?

File: src/posthog-extended.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+ * Anything that is considered "extended" should be here. Core functionality is only around standard capture and feature flags logic
+ * @constructor
+ */
+export class PostHogExtended extends PostHogCore {
+    protected _Cls = PostHogExtended
+    surveys: PostHogSurveys
+    toolbar: Toolbar
+    autocapture?: Autocapture
+    sessionRecording?: SessionRecording
+    webPerformance = new DeprecatedWebPerformanceObserver()
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply src/posthog-extended.ts:41]: Yeah its a beefy bit of js so 🤷

[the remark this answers is not in the document]
Comment (pauldambra) [reply src/posthog-extended.ts:41]: No, it's a good shout... I like it. folk want a tiny bundle let's give them the option.

**Pull request closed without merging**